File: ConsoleApplication/Tree/NodeInsertion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApplication1.Tree
{
    public class NodeInsertion
    {
        public void Insert()
        {
            TreeNode root = Tree.GetExampleTree2();

            TreeNode result = InsertRec(root, 25);

        }

        private TreeNode InsertRec(TreeNode root, int val)
        {
            if (root == null) return new TreeNode(val);

            if (root.Data > val)
            {
                root.Left = InsertRec(root.Left, val);
            }
            else if (root.Data < val)
            {
                root.Right = InsertRec(root.Right, val);
            }

            return root;
        }
    }
}

File: Tasks/src/trees/NodeInsertion.kt
package trees

import trees.model.Tree
import trees.model.TreeNode

class NodeInsertion {
    fun insert() {
        val root = Tree.getExampleTree1()
        insertNode(root, 43)
    }

    private fun insertNode(root: TreeNode?, value: Int): TreeNode {
        if (root == null) return TreeNode(value)

        if (root.value > value) {
            root.left = insertNode(root.left, value)
        } else if (root.value < value) {
            root.right = insertNode(root.right, value)
        }

        return root
    }
}